tiledb/sm/metadata/metadata.cc

```diff
@@ -198,7 +198,9 @@ Status Metadata::del(const char* key) {
 
   MetadataValue value;
   value.del_ = 1;
-  metadata_map_.emplace(std::make_pair(std::string(key), std::move(value)));
+  std::string key_str(key);
+  metadata_map_.erase(key_str);
+  metadata_map_.emplace(std::make_pair(key_str, std::move(value)));
```

Comment:
You can use [`insert_or_assign`](https://en.cppreference.com/w/cpp/container/map/insert_or_assign) to avoid separate delete (lookup) and emplace.

Reply:
So that's the function. Thanks.